spring-shell2-core/src/main/java/org/springframework/shell2/standard/StandardParameterResolver.java

```diff
@@ -98,7 +98,7 @@ public void setValueProviders(Collection<ValueProvider> valueProviders) {
 
 	@Override
 	public boolean supports(MethodParameter parameter) {
-		return true;
+		return parameter.getMethodAnnotation(ShellMethod.class) != null;
```

Comment:
But the method still needs to be annotated, right? I'm not checking for the ShellOption but for the ShellMethod one. (I'm checking the StandardMethodTargetResolver::resolve method)

This was to allow having the StandardParameterResolver and the Legacy one loaded in the same runtime without colliding.